modules/hitl/src/views/full/sidebar/index.jsx

```diff
@@ -55,29 +67,73 @@ export default class Sidebar extends React.Component {
 
   render() {
     const filterTooltip = <Tooltip id="tooltip">Show only paused conversations</Tooltip>
+    const searchTooltip = <Tooltip id="tooltip">Search By Name</Tooltip>
+    const searchClearTooltip = <Tooltip id="tooltip">Clear Search</Tooltip>
     const filterStyle = {
       color: this.props.filter ? '#56c0b2' : '#666666'
     }
+    const cancelsearch_style = {
+      left: '90px'
+    }
 
     const dynamicHeightUsersDiv = {
       height: innerHeight - 160
     }
 
     return (
       <div className={style.sidebar}>
-        <div className={style.header}>
+        <div style={{ display: this.props.customSearchProps.searchStatus ? 'none' : 'block' }} className={style.header}>
           <div className={style.filter}>
-            <OverlayTrigger placement="top" overlay={filterTooltip}>
+            <OverlayTrigger placement="bottom" overlay={filterTooltip}>
               <i className="material-icons" style={filterStyle} onClick={this.toggleFilter}>
                 bookmark
               </i>
             </OverlayTrigger>
           </div>
+          <div className={style.filter + ' ' + style.icon_right}>
+            <OverlayTrigger placement="bottom" overlay={searchTooltip}>
+              <i className="material-icons" style={filterStyle} onClick={this.toggleSearch}>
+                search
+              </i>
+            </OverlayTrigger>
+          </div>
+          <div style={{ float: 'right' }} className={style.filter} style={cancelsearch_style}>
+            <OverlayTrigger placement="bottom" overlay={searchClearTooltip}>
+              <i className="material-icons" style={filterStyle} onClick={this.clearSearch}>
+                clear_all
+              </i>
+            </OverlayTrigger>
+          </div>
         </div>
+
+        <div style={{ display: this.props.customSearchProps.searchStatus ? 'block' : 'none' }} className={style.header}>
+          <div className={style.textfilter}>
+            <FormControl
+              value={this.props.customSearchProps.searchText}
+              onChange={this.handleChange}
+              placeholder="Search By Name"
+            />
+          </div>
+          <div className={style.filter + ' ' + style.text_right}>
+            <OverlayTrigger placement="bottom" overlay={searchTooltip}>
+              <i className="material-icons" style={filterStyle} onClick={this.toggleSearch}>
+                search
+              </i>
+            </OverlayTrigger>
+          </div>
+          <div className={style.filter + ' ' + style.cancel_right}>
+            <OverlayTrigger placement="bottom" overlay={searchClearTooltip}>
+              <i className="material-icons" style={filterStyle} onClick={this.clearSearch}>
+                cancel
+              </i>
+            </OverlayTrigger>
+          </div>
+        </div>
+
```

Comment:
I did not see the module interface, but if all your logic for searching is in the sidebar, wouldn't it make more sense to keep all the state here (instead of passing your search props), then just calling the parent to do the actual search? (eg: this.props.executeSearch(this.state.searchQuery)

Reply:
[Done](https://github.com/shamoh19/botpress/commit/3c171a8926b82c779d9f24997da72b05e98cd396#diff-aea332c672541f3ad09f68a0c3dddcd4)